components/admin/users/AdminUsersSplitSearch.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import _ from 'lodash'
+
+const split: any = (
```

Comment:
hahahaah its so amazing you wrote your own split function haha. there is `.split` function that you should just use.

```
"hello".split('ell') // ['h', 'o']
```

Reply:
My split function will include the delimiter.
split('hello', 'ell') // ['h', 'ell', 'o']
The delimiter is needed so that `line 54` (`const value = splitArr.map((word: string) => {`)
can highlight the searchTerm

Reply:
Can't this be done using regex and replace? Like `"hello".replace(/ell/g, "<span style={{ backgroundColor: bgColor }}>ell</span>")`

Reply:
@ggwadera using `span` tag in quotes as a string instead of react component makes it inconsistent.

@anthonykhoa I see your point, perhaps you could use the [`_.join` function from lodash](https://lodash.com/docs/4.17.15#join)

```
const splitStr = _.join(str.toLowerCase().split(....), ...)
```

Reply:
I didn't think to use join. That would make my code much cleaner. Thanks!

Reply:
@songz I am experiencing trouble using `_.join`. I need to use an array, but using `_.join` returns a string. I will try to think of other logic to implement.

Reply:
@anthonykhoa 

```
'hello'.split('ell').reduce( (acc, e, i) => {
  acc.push( <span>{e} </span> )
  if (i < 'hello'.length - 1) {
    arr.push( <span className="matchStr">ell</span> )
  }
  return acc
})
```